### PR TITLE
migrated from mocha to vitest

### PR DESCRIPTION
**Pull request by jonah-iden**
test now run via vitest. Otherwise there should have not been much change.
- there is now a `yarn coverage` command to chekc the test coverage
- tests can both be executed from root to execute all or for single packages
- there are debug targets for executing all tests and executing single test files

### CODE REVIEW

File: packages/sprotty/tsconfig.json [hunk 1]
```diff
@@ -8,5 +8,7 @@
   ],
   "exclude": [
     "**/*.spec.ts", "**/*.spec.tsx", "src/utils/test-helper.ts"
-  ]
+  ],
+  "experimentalDecorators": true,
+  "emitDecoratorMetadata": true
```
Comment (spoenemann) [inline packages/sprotty/tsconfig.json:13]: Shouldn't these be part of the `compilerOptions`? The base tsconfig should already include them.

**Review by spoenemann — APPROVED** [review]
Thanks! Just one detail

File: packages/sprotty/tsconfig.json (hunk 1 above)
Comment (jonah-iden) [reply packages/sprotty/tsconfig.json:13]: ups yeah, i think forgot to remove them again. I'll do that right away

**Comment by dhuebner** [comment]
@jonah-iden 
I have in some spec files eslint errors: e.g. 'missing semicolon' in diagram-server.spec.ts. Are thees new because of eslint configuration changes? 
An additional question: Is it possible with vitest to somehow `import 'reflect-metadata';` by default for all tests?

**Comment by jonah-iden** [comment]
@dhuebner these eslint errors seem to have been there before, they are also there on master. I didn't change anything realted to eslint.
Regarding your second question, i searched for a solution while implementing this, but sadly couldn't find anything. If you know of any i would be happy to add that

**Comment by spoenemann** [comment]
Currently we exclude test files when we execute eslint:
https://github.com/eclipse-sprotty/sprotty/blob/727c2246715fc633be1e3cc65bfcdf4fce9ed359/packages/sprotty/package.json#L50

But the ESLint extension shows problem markers for the test files anyway. I propose to remove the exclusion of test files and apply eslint to the whole code base. WDYT?

**Comment by dhuebner** [comment]
@jonah-iden 
Thank you for clarification! 

@spoenemann 
I think it is a good idea to generalize the eslint settings. One problem I see, is that when writing test code there are sometime situations (unused parameters, reflection) where one needs to write code that is then marked as error. That why I think it would be nice to have a separate config file for *.spec*.ts where we can overwrite some settings.  
I also noticed that when working with examples no checks are reported in the editor but the build fails then with e.g. unused imports ` TS6133: 'Action' is declared but its value is never read.`.

**Comment by spoenemann** [comment]
Separate config files don't work well, the editor will always decide for one config. What we can do is add _ignore_ comments where needed to suppress the eslint errors/warnings.

**Comment by kaisalmen** [comment]
@jonah-iden @dhuebner and @spoenemann most promising approach with workspaces and eslint so far I had was to you have one specific tsconfig (or shared one at workspace top with vscode) for eslint that includes all code from all packages and then have only one eslint script in the root that scans everything ([example](https://github.com/TypeFox/monaco-components/blob/main/.eslintrc.cjs))

**Comment by spoenemann** [comment]
@kaisalmen @jonah-iden or @dhuebner, would one of you like to create a PR for that?

**Comment by kaisalmen** [comment]
✋

**Comment by kaisalmen** [comment]
See https://github.com/eclipse-sprotty/sprotty/pull/400

**Pull request merged**